Movie Page API

KrystianGreblowski opened the pull request:
I have added data from API to MoviePage and I have also added many different things to make everything work well.

KrystianGreblowski committed:
folders and files reorganization for popular people page

KrystianGreblowski committed:
example Header deleted

KrystianGreblowski committed:
example Header deleted

KrystianGreblowski committed:
example image for cast and crew tiles added

KrystianGreblowski committed:
Title changed to TilesHeader and paths to TilesHeader, TilesContainer and PersonTile changed

KrystianGreblowski committed:
code formatted

KrystianGreblowski committed:
max-width changed to 1400px and padding: 0 16px added

KrystianGreblowski committed:
App changed for displaying only MoviePage (only for a while)

KrystianGreblowski committed:
breakpoint deleted

KrystianGreblowski committed:
max-width deleted

KrystianGreblowski committed:
background-color changed to black

KrystianGreblowski committed:
margin-bottom added

KrystianGreblowski committed:
a few changes to styled Banner

KrystianGreblowski committed:
query language added

KrystianGreblowski committed:
getMovieDetails added

KrystianGreblowski committed:
movieDetailsSlice added

KrystianGreblowski committed:
movieDetailsSaga added

KrystianGreblowski committed:
alt in PersonTile changed

KrystianGreblowski committed:
movie details init added (for example movie)

KrystianGreblowski committed:
data from Api for Banner added

KrystianGreblowski committed:
typo mistake

KrystianGreblowski committed:
loading and error state added

KrystianGreblowski committed:
loading and error state added

KrystianGreblowski committed:
a few changes added

KrystianGreblowski committed:
credits to api request added (using append_to_response)

KrystianGreblowski committed:
gap changed from 6px to 8px

KrystianGreblowski committed:
api data for MainInfo added

KrystianGreblowski committed:
extraInfo added

KrystianGreblowski committed:
extraInfo styles added

KrystianGreblowski committed:
nanoid added

KrystianGreblowski committed:
Loading and Error handling moved to MoviePage/index.js

KrystianGreblowski committed:
image moved to other folder

KrystianGreblowski committed:
Cast and Crew data from Api added

KrystianGreblowski committed:
console.log deleted

KrystianGreblowski committed:
banner will not be displayed if backdrop_path === null

KrystianGreblowski committed:
department changed to job

KrystianGreblowski committed:
woodsmoke color changed to black

KrystianGreblowski committed:
TilesButton deleted

KrystianGreblowski committed:
TilesButton deleted

KrystianGreblowski committed:
Movie Page added to router

KrystianGreblowski committed:
MovieTileButton added

KrystianGreblowski committed:
fetchMovieDetailsInit deleted

KrystianGreblowski committed:
movie page displaying after clicking a movie tile - added

KrystianGreblowski committed:
NavLink added

KrystianGreblowski committed:
Movies Details example page deleted

KrystianGreblowski committed:
color: inherit added

KrystianGreblowski committed:
unnecessary images deleted

KrystianGreblowski committed:
fetchCurrentPage renamed to fetchCurrentMoviePage

KrystianGreblowski committed:
fetchCurrentPage changed to fetchCurrentPeoplePage

KrystianGreblowski committed:
missing "s" added

KrystianGreblowski committed:
watchFetchMovieTypes renamed to watchFetchMovieTypesInit

KrystianGreblowski committed:
watchFetchMovieDetails renamed to watchFetchMovieId

KrystianGreblowski committed:
LoadingPage and ErrorPage examples deleted

KrystianGreblowski committed:
box-shadow inset added

KrystianGreblowski committed:
LoadingBanner added

KrystianGreblowski committed:
Shadow.png deleted

KrystianGreblowski committed:
ScrollToTop component added

KrystianGreblowski committed:
return null added to ScrollToTop

KrystianGreblowski committed:
loading state placeholders added

KrystianGreblowski committed:
maxNumberOfTiles added to Array constructor

KrystianGreblowski committed:
state-status = "loading" added to fetchMovieId

KrystianGreblowski committed:
localStorage for movie details added

KrystianGreblowski committed:
console.log deleted

KrystianGreblowski committed:
MovieTileButton deleted (it was no longer necessary)

KrystianGreblowski committed:
Banner styles corrected

KrystianGreblowski committed:
MainInfo styles corrected

KrystianGreblowski committed:
flex-end changed to flex-start

Alekzann commented on src/features/MoviePage/Content/MainInfo/styled.js:
Nice decision with grid, as you said previously! Good job!

Alekzann reviewed (changes requested):
It was a really great and excellent job! Unfortunately I can't appreciate the work with slice and saga, but the main thing is that it works! 
But, shadow in MoviePage in poster should be bigger, because when we have light image it's hard to see the movie title, crew should includes more persons.
Distance between MainInfo and Poster too big (as for me).

KrystianGreblowski committed:
margin changed to 64px

KrystianGreblowski committed:
number of Crew tiles changed to 12

KrystianGreblowski committed:
margin changed under mobile breakpoint

KrystianGreblowski commented:
Thanks for the review. I changed the distance between MainInfo and Banner. I also changed the number of crew tiles. In my opinion, the box shadow is good enough and can't be bigger because it would take up too much space.

Alekzann commented on src/features/MoviePage/Banner/styled.js:
```diff
@@ -3,30 +3,29 @@ import styled from "styled-components";
-  top: 0;
-  left: 0;
+  max-height: 770px;
+  box-shadow: inset 0 0 100px 100px rgba(0, 0, 0);
```
What do you think about small changing of box-shadow from this to 
box-shadow: inset 0px 0px **150px 140px** rgba(0, 0, 0);
At least for bigger displays.

Alekzann reviewed (approved):
Great job!
